arduino/smartcar/smartcar.ino

```diff
@@ -14,18 +18,77 @@ void setup() {
 }
 
 void loop() {
-  car.setSpeed(90);
-  preventCrash();
+  car.setSpeed(50); 
+  preventCrashWithUltrasonic(); 
+  preventCrashWithInfraredFront(); 
+  preventCrashWithInfraredLeft();
+  preventCrahsWithInfraredRight();
+  preventCrashWithInfraredBack();
 }
 
-void preventCrash() {
-  unsigned int fDistance = front.getDistance();
+void preventCrashWithUltrasonic() {
+  unsigned int fDistance = frontUS.getDistance();
   if (fDistance > 0 && fDistance < 200) {
+    Serial.print("Obstacle detected by front ultrasonic sensor");
     car.setSpeed(0);
     delay(500);
-    control.overrideMotorSpeed(100, -100);
+    control.overrideMotorSpeed(50, -100);
     delay(800);
     car.setSpeed(0);
     delay(200);
   }
 }
+
+void preventCrashWithInfraredFront() {
+  unsigned int fDistance = frontIR.getDistance();
+  if (fDistance > 0 && fDistance < 200) {
+    Serial.print("Obstacle detected by front infrared sensor"); 
+    car.setSpeed(0);
+    delay(500);
+    control.overrideMotorSpeed(50, -100);
+    delay(800);
+    car.setSpeed(0);
+    delay(200);
+  }
+}
+
+void preventCrashWithInfraredLeft() {
+  unsigned int fDistance = leftIR.getDistance();
+  if (fDistance > 0 && fDistance < 200) {
+    Serial.print("Obstacle detected by left infrared sensor"); 
+    car.setSpeed(0);
+    delay(500);
+    control.overrideMotorSpeed(50, -100);
+    delay(800);
+    car.setSpeed(0);
+    delay(200);
+  }
+}
```

Comment:
What good is this method actually? As far as I can tell, when the car is passing by a tree on its left side that it is absolutely not on a collision course with, it will stop and turn around. I can see the value of detecting an obstacle to the sides, for example to prevent turning in that direction at a particularly sharp angle, where the front sensor might not detect the obstacle in time. But if there is no imminent crash the car should not be stopped.